angular-ngrx-scss/src/app/state/profile/profile.selectors.ts

```diff
@@ -26,25 +26,41 @@ export const selectFilterBySearch = createSelector(
   (state: ProfileState) => state.sortAndFilter?.search,
 );
 
+export const selectFilterByType = createSelector(
+  selectProfileState,
+  (state: ProfileState) => state.sortAndFilter?.type || TypeFilter.All,
+);
+
 export const hasActiveSortAndFilters = createSelector(
   selectFilterBySearch,
-  Boolean,
+  selectFilterByType,
+  (search, type) => !!search || type !== TypeFilter.All,
 );
 
 export const selectRepos = createSelector(
   selectProfileState,
   selectFilterBySearch,
-  (state: ProfileState, search?: string) => {
+  selectFilterByType,
+  (state: ProfileState, search?: string, type?: string) => {
+    let filteredRepos = state.repos;
     if (search) {
-      return state.repos?.filter((item) =>
+      filteredRepos = filteredRepos?.filter((item) =>
         item.name.toLowerCase().includes(search.toLowerCase()),
       );
     }
-    return state.repos ?? [];
+    if (type !== TypeFilter.All) {
+      if (type === TypeFilter.Archived) {
+        filteredRepos = filteredRepos?.filter((item) => item.archived === true);
+      }
+      if (type === TypeFilter.Forked) {
+        filteredRepos = filteredRepos?.filter((item) => item.fork === true);
```

Comment:
suggestion: We can just use the `Boolean` constructor here rather than wrapping things in an arrow function.

```suggestion
        filteredRepos = filteredRepos?.filter(Boolean);
```

Reply:
`filteredRepos = filteredRepos?.filter((item) => Boolean(item.fork));`

same here.